src/ucp/tag/rndv.c

```diff
@@ -104,8 +96,16 @@ static size_t ucp_tag_rndv_rtr_pack(void *dest, void *arg)
 
     /* Pack remote keys (which can be empty list) */
     if (UCP_DT_IS_CONTIG(rreq->recv.datatype)) {
+        /* Adjust the real receive length by the length that is expected
+         * to be sent (i.e. it has to be minimum between the length of
+         * the receive buffer and the length specified in the RTS header)
```

Comment:
this looks like invalid assumption. It is possible that receive and send buffer lengths will be different. And `ucp_tag_rndv_rtr_pack` takes the length of receive buffer

Reply:
why? a receiver registers minimum between receive size and send size (from RTS)
we have to ask send to us only send size rather than recv size (if they are not equal)